pkg/controller/pxc/controller.go

```diff
@@ -272,6 +272,19 @@ func (r *ReconcilePerconaXtraDBCluster) Reconcile(ctx context.Context, request r
 		}
 	}
 
+	if o.Spec.ProxySQLEnabled() {
+		haproxySts := appsv1.StatefulSet{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      o.Name + "-haproxy",
+				Namespace: o.Namespace,
+			},
+		}
+		err = r.client.Get(ctx, client.ObjectKeyFromObject(&haproxySts), &haproxySts)
```

Comment:
What if the user starts the cluster from old PVCs or the cluster is paused? It may not be enough to check haproxySts. 
Also, I am worried about the update from 1.12 to 1.13. As I understand, we will set sha2, and if a user creates a new user in db, it will be used, but the user's current infrastructure could not be ready for it. I am not sure that we need to change the plugin during the update automatically.

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/1415/commits/150ef81a527f9f2426c5252a2a52324b1dc68fe8